.github/workflows/compiler-warnings.yml

```diff
@@ -24,7 +24,12 @@ jobs:
       ISPC_OPAQUE_PTR_MODE: "ON"
 
     steps:
-    - uses: actions/checkout@v4
+    - name: Harden Runner
+      uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
```

Comment:
Do we really need it? It looks like it tracks and/or interferes with the whole build process afterwards (network, filesystem operations, processes and etc.). I am just worried that this, in itself, may be a greater threat than the potential threats it seeks to prevent.

Reply:
I'll double check it with security guys.

Reply:
From a security standpoint, the Harden Runner job is considered safe. However, during its audit job, it didn't uncover anything useful. Many of its checks simply repeat the OpenSSF scores, which we already adhere to. As a result, it doesn't bring any added value for us.

With Step-Security, it's easier to create a new PR instead of modifying the existing one. Therefore, here is a new PR without the hardening runners: https://github.com/ispc/ispc/pull/2916